HealthcareApp/src/MigrationAdminPanel/Configuration/HostBuilderConfigurator.cs

```diff
@@ -0,0 +1,51 @@
+using Application.Abstractions;
+using Application.Abstractions.Decorators;
+using Domain.Entities;
+using Healthcare.Infrastructure.Persistance;
+using Infrastructure.Repository;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+using MigrationAdminPanel.Services;
+
+namespace MigrationAdminPanel.Configuration;
+
+public class HostBuilderConfigurator
+{
+    public static IHostBuilder CreateHostBuilder(string[] args) =>
+        Host.CreateDefaultBuilder(args)
+            .ConfigureServices((context, services) =>
+            {
+                var builder = new ConfigurationBuilder()
+                    .SetBasePath(Directory.GetCurrentDirectory())
+                    .AddJsonFile("appsettings.json", optional: false);
+                
+                IConfiguration config = builder.Build();
+                
+                // Register DbContext with a connection string
+                services.AddDbContext<AppDbContext>(options =>
+                {
+                    options.UseSqlServer(config.GetConnectionString("HealthcareDb"));
+                    options.UseQueryTrackingBehavior(QueryTrackingBehavior.NoTracking);
```

Comment:
It can be tricky to understand that entities are not tracked. I suggest removing it and adding AsNoTracking individually for each method where you don't need to track entities.